blaze/server/spider.py

```diff
@@ -72,28 +73,47 @@ def data_spider(path,
     """
     # NOTE: this is named `data_spider` rather than just `spider` to
     # disambiguate this function from the `blaze.server.spider` module.
-    return {
-        os.path.basename(path): _spider(path, ignore=ignore,
-                                        followlinks=followlinks,
-                                        hidden=hidden,
-                                        extra_kwargs=extra_kwargs)
-    }
+    return {os.path.basename(path): _spider(path,
+                                            ignore=ignore,
+                                            followlinks=followlinks,
+                                            hidden=hidden,
+                                            extra_kwargs=extra_kwargs)}
 
 
-def from_yaml(path, ignore=(ValueError, NotImplementedError), followlinks=True,
-              hidden=False):
+@contextmanager
+def manage_chdir(root):
```

Comment:
I think an easier name for this might be `pushd`. `manage` doesn't quite tell me how it is being managed but the pushd/popd analog might be more clear